bindings/wasm/src/storage/key_id_storage.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::WasmMethodDigest;
+use crate::common::PromiseString;
+use crate::error::JsValueResult;
+use identity_iota::storage::key_id_storage::KeyIdStorage;
+use identity_iota::storage::key_id_storage::KeyIdStorageResult;
+use identity_iota::storage::key_id_storage::MethodDigest;
+use identity_iota::storage::key_storage::KeyId;
+use js_sys::Promise;
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen_futures::JsFuture;
+
+#[wasm_bindgen]
+extern "C" {
+  #[wasm_bindgen(typescript_type = "KeyIdStorage")]
+  pub type WasmKeyIdStorage;
+
+  #[wasm_bindgen(method, js_name = insertKeyId)]
+  pub fn insert_key_id(this: &WasmKeyIdStorage, method_digest: WasmMethodDigest, key_id: String) -> Promise;
+
+  #[wasm_bindgen(method, js_name = getKeyId)]
+  pub fn get_key_id(this: &WasmKeyIdStorage, method_digest: WasmMethodDigest) -> PromiseString;
+
+  #[wasm_bindgen(method, js_name = deleteKeyId)]
+  pub fn delete_key_id(this: &WasmKeyIdStorage, method_digest: WasmMethodDigest) -> Promise;
```

Comment:
Either add a comment explaining that we must ensure that the `method_digest` argument is never passed in from JS because then they will get nulled out by Rust, or make the methods in this block (but not in the `KeyIdStorage` trait) take `&WasmMethodDigest`.

Reply:
Taking a reference here isn't allowed by bindgen. For now, I would leave it since it's not used and only there for eventual future need, the same thing exists for `WasmJwkStorage`.